test_unit.py

```diff
@@ -1209,6 +1209,35 @@ def test_HTLC(self):
         self.assertEqual(p, base64.b64decode(golden))
         self.assertEqual(s.get_address(), golden_addr)
 
+    def test_dynamic_fee(self):
+        addr1 = "726KBOYUJJNE5J5UHCSGQGWIBZWKCBN4WYD7YVSTEXEVNFPWUIJ7TAEOPM"
+        addr2 = "42NJMHTPFVPXVSDGA6JGKUV6TARV5UZTMPFIREMLXHETRKIVW34QFSDFRE"
+        s = template.DynamicFee(addr1, 5000, 12345, 12346, addr2)
+        s.lease_value = base64.b64decode("f4OxZX/x/FO5LcGBSKHWXfwtSx+j1ncoSt3SABJtkGk=")
+
+        golden_addr = ("GCI4WWDIWUFATVPOQ372OZYG52EUL" +
+                       "PUZKI7Y34MXK3ZJKIBZXHD2H5C5TI")
+
+        golden = ("ASAFAgGIJ7lgumAmAyD+vKC7FEpaTqe0OKRoGsgObKEFvLYH/FZTJclW" +
+                  "lfaiEyDmmpYeby1feshmB5JlUr6YI17TM2PKiJGLuck4qRW2+SB/g7Fl" +
+                  "f/H8U7ktwYFIodZd/C1LH6PWdyhK3dIAEm2QaTIEIhIzABAjEhAzAAcx" +
+                  "ABIQMwAIMQESEDEWIxIQMRAjEhAxBygSEDEJKRIQMQgkEhAxAiUSEDEE" +
+                  "IQQSEDEGKhIQ")
+        p = s.get_program()
+        self.assertEqual(p, base64.b64decode(golden))
+        self.assertEqual(s.get_address(), golden_addr)
+        sk, pk = account.generate_account()
+        txn, lsig = s.sign_dynamic_fee(sk, "f4OxZX/x/FO5LcGBSKHWXfwtSx+j1ncoSt3SABJtkGk=")
+        txns = s.get_transactions(txn, lsig, sk, 10, 1234, 2234)
+
+        golden_txn_1 = transaction.PaymentTxn(pk, 10, 12345, 12346, "f4OxZX/x/FO5LcGBSKHWXfwtSx+j1ncoSt3SABJtkGk=", addr1, 5000, close_remainder_to=addr2, lease=s.lease_value)
+        golden_txn_1.group = txn.group
+        self.assertEqual(golden_txn_1, txns[0].transaction)
+
+        golden_txn_2 = transaction.PaymentTxn(pk, 10, 1234, 2234, "f4OxZX/x/FO5LcGBSKHWXfwtSx+j1ncoSt3SABJtkGk=", pk, golden_txn_1.fee, lease=s.lease_value)
+        golden_txn_2.group = txn.group
+        self.assertEqual(golden_txn_2, txns[1].transaction)
```

Comment:
Sorry, one more thing. These golden transactions should be an imported string so that we can compare them across SDKs.